twisted/web/test/test_http2.py

```diff
@@ -2172,3 +2173,206 @@ def test_initiallySchedulesOneDataCall(self):
         self.assertEqual(call.func, a._sendPrioritisedData)
         self.assertEqual(call.args, ())
         self.assertEqual(call.kw, {})
+
+
+
+class HTTP2TimeoutTests(unittest.TestCase):
+    """
+    The L{H2Connection} object times out idle connections.
+    """
+    if skipH2:
+        skip = skipH2
+
+
+    getRequestHeaders = [
+        (b':method', b'GET'),
+        (b':authority', b'localhost'),
+        (b':path', b'/'),
+        (b':scheme', b'https'),
+        (b'user-agent', b'twisted-test-code'),
+        (b'custom-header', b'1'),
+        (b'custom-header', b'2'),
+    ]
+
+
+    def test_timeoutAfterInactivity(self):
+        """
+        When a L{H2Connection} does not receive any data for more than the
+        time out interval, it closes the connection cleanly.
+        """
+        reactor = task.Clock()
+        conn = H2Connection(reactor)
+        conn.timeOut = 100
+
+        # This monkeypatch is needed because TimeoutMixin doesn't accept being
+        # passed a reactor.
+        conn.callLater = reactor.callLater
+
+        f = FrameFactory()
+        b = StringTransport()
```

Comment:
why `b` for `transport`? why not just call this variable `transport`. For me, this makes the test easier to read... I am just not into the theoretical mathematics and a/b/c, n/p/r and x/y/z aren not my friends :)